sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -581,42 +588,38 @@ public GlobalThroughputControlConfigBuilder createGlobalThroughputControlConfigB
         return new GlobalThroughputControlConfigBuilder(this, databaseId, containerId);
     }
 
+    WriteRetryPolicy getNonIdempotentWriteRetryPolicy() {
+        return this.nonIdempotentWriteRetryPolicy;
+    }
+
     void openConnectionsAndInitCaches() {
-        blockListVoidResponse(openConnectionsAndInitCachesInternal());
+        blockVoidFlux(asyncDocumentClient.submitOpenConnectionTasksAndInitCaches(proactiveContainerInitConfig));
     }
 
-    WriteRetryPolicy getNonIdempotentWriteRetryPolicy() {
-        return this.nonIdempotentWriteRetryPolicy;
+    void openConnectionsAndInitCaches(Duration aggressiveProactiveConnectionEstablishmentDuration) {
+        Flux<Void> submitOpenConnectionTasksFlux = asyncDocumentClient.submitOpenConnectionTasksAndInitCaches(proactiveContainerInitConfig);
+
+        wrapSourceFluxAndSoftCompleteAfterTimeout(
+                submitOpenConnectionTasksFlux,
+                aggressiveProactiveConnectionEstablishmentDuration,
+                CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC)
+                .blockLast();
     }
 
-    private Mono<List<Void>> openConnectionsAndInitCachesInternal() {
-        int concurrency = 1;
-        int prefetch = 1;
-        if (this.proactiveContainerInitConfig != null) {
-            return Flux.fromIterable(this.proactiveContainerInitConfig.getCosmosContainerIdentities())
-                    .flatMap(
-                        cosmosContainerIdentity -> Mono.just(this
-                            .getDatabase(containerIdentityAccessor.getDatabaseName(cosmosContainerIdentity))
-                            .getContainer(containerIdentityAccessor.getContainerName(cosmosContainerIdentity))),
-                        concurrency,
-                        prefetch
-                    )
-                    .flatMap(
-                        cosmosAsyncContainer -> cosmosAsyncContainer
-                            .openConnectionsAndInitCaches(
-                                this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount()),
-                        concurrency,
-                        prefetch)
-                    .collectList();
-        }
-        return Mono.empty();
+    private <T> Flux<T> wrapSourceFluxAndSoftCompleteAfterTimeout(Flux<T> source, Duration timeout, Scheduler executionContext) {
+        return Flux.<T>create(sink -> {
+                    source
+                            .subscribeOn(executionContext)
+                            .subscribe(t -> sink.next(t));
```

Comment:
Doesn't the `Flux#create` factory just buffer all signals if downstream can't keep up? Losing signals is not an issue now since the upstream returns a Flux<Void>, it could be when we want to process published open connection results for diagnostic / logging purposes.